lib/client/client.js

```diff
@@ -1005,6 +1013,8 @@ Client.prototype.connect = function connect () {
     // Communicate the last-encountered error
     if (err instanceof ConnectionError) {
       self.emit('connectTimeout', err)
+    } else if (err.code === 'ECONNREFUSED') {
+      self.emit('connectRefused', err)
     } else {
       self.emit('error', err)
```

Comment:
since this only emits errors in `connect` should this just be changed to emit 'connectError'?

FYI I believe this could be a breaking change.

Reply:
Those errors already emit a `connectError` event on https://github.com/ldapjs/node-ldapjs/blob/5afa6772099f11b36c92c677c73bd483ad2d7c80/lib/client/client.js#L803
This will emit another (more specific) error event.

On one hand, this can be a breaking change for codes handling connection errors with the `error` event. On the other side, the callback for `bind` will be called with the error so previously written code will still work.

Reply:
I think I am okay with the breaking change here since it will be going into a new major version. I have stated that the _intention_ is to not drastically alter the API, but some breaking changes are going to make it in.